platform-modules/batch-models/src/main/scala/org/ekstep/analytics/updater/UpdateContentRating.scala

```diff
@@ -0,0 +1,72 @@
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.ekstep.analytics.framework._
+import org.ekstep.analytics.framework.conf.AppConf
+import org.joda.time.DateTime
+import org.ekstep.analytics.framework.util.{HTTPClient, JobLogger, RestUtil}
+
+case class ContentRatingResult(contentId: String, averageRating: Double) extends AlgoOutput with Output
+case class Response(id: String, ver: String, ts: String, params: Params, responseCode: String, result: Map[String, String])
+
+object UpdateContentRating  extends IBatchModelTemplate[Empty, Empty, ContentRatingResult, ContentRatingResult] with Serializable {
+
+  implicit val className = "org.ekstep.analytics.updater.UpdateContentRating"
+  override def name: String = "UpdateContentRating"
+
+  override def preProcess(data: RDD[Empty], config: Map[String, AnyRef])(implicit sc: SparkContext): RDD[Empty] = {
+      data
+  }
+
+  override def algorithm(data: RDD[Empty], config: Map[String, AnyRef])(implicit sc: SparkContext): RDD[ContentRatingResult] = {
+
+      val contentList = getRatedContents(config, RestUtil)
+      val contentRatingList = getContentRatings(RestUtil)
+      val finalList = contentRatingList.filter(f => contentList.contains(f.contentId))
+      JobLogger.log("content-ids: " + contentList.size + " rating list: " + contentRatingList.size + " filtered list: " + finalList.size, None, Level.INFO)
+      sc.parallelize(finalList)
+  }
+
+  override def postProcess(data: RDD[ContentRatingResult], config: Map[String, AnyRef])(implicit sc: SparkContext): RDD[ContentRatingResult] = {
+      val baseURL = AppConf.getConfig("lp.system.update.base.url")
+      if (data.count() > 0) {
+          data.foreach { f =>
+              val response = publishRatingToContentModel(f.contentId, f.averageRating, baseURL, RestUtil)
+              JobLogger.log("System Update API request for " + f.contentId + " is " + response.params.status.getOrElse(""), None, Level.INFO)
+          }
+      }
+      data
+  }
+
+  def getRatedContents(config: Map[String, AnyRef], restUtil: HTTPClient): List[String] = {
+      val apiURL = AppConf.getConfig("druid.sql.host")
+      val startDate = config.getOrElse("startDate", new DateTime().minusDays(1).toString("yyyy-MM-dd"))
+      val endDate = config.getOrElse("endDate", new DateTime().toString("yyyy-MM-dd"))
+      val contentRequest = AppConf.getConfig("druid.unique.content.query").format(new DateTime(startDate).getMillis, new DateTime(endDate).getMillis)
+      val contentResponse = restUtil.post[List[Map[String, AnyRef]]](apiURL, contentRequest)
+      contentResponse.map(x => x.getOrElse("Id", "").toString)
+  }
+
+  def getContentRatings(restUtil: HTTPClient): List[ContentRatingResult] = {
+      val apiURL = AppConf.getConfig("druid.sql.host")
+      val ratingRequest = AppConf.getConfig("druid.content.rating.query")
+      val contentRatingResponse = restUtil.post[List[Map[String, AnyRef]]](apiURL, ratingRequest)
```

Comment:
Please modify the query to aggregate only on the content list which we got for yesterday instead on all the content in Druid. Please try with `IN` operator and check the performance